src/API.jl

```diff
@@ -20,14 +20,21 @@ include("generate.jl")
 
 function check_package_name(x::AbstractString)
     if !(occursin(Pkg.REPLMode.name_re, x))
-         pkgerror("$x is not a valid packagename")
+        islikelyurl(x) = any(occursin.(['\\','/'], x))
+        message = "$x is not a valid packagename."
+        if islikelyurl(x)
```

Comment:
It seems like it would be just as clear to replace `islikelyurl(x)` with its definition from two lines above, no? If you feel it needs more explanation, you can put a comment along the lines of `# not a valid package name, possible path or URL`.

Reply:
I'm personally a fan of the self-documenting code philosophy, but I'll defer to you both.

Reply:
Taking that to the logical conclusion would mean that any code that requires a comment should be factored out into a function with the comment as its name...

Reply:
You're not wrong. It's a balance of course, but I find aiming for that makes it easier to understand later what I was doing (coming from C#). Reserves comments for explaining why choices were made, don't have to keep comments and code in sync as much, makes refactoring a bit easier. IMO.